app/services/color_scheme_revisor.rb

```diff
@@ -20,6 +20,18 @@ def revise(update_existing_colors_only: false)
       @color_scheme.user_selectable = @params[:user_selectable] if @params.has_key?(
         :user_selectable,
       )
+
+      if @params.has_key?(:default_light_on_theme)
+        Theme.find_default.update!(
+          color_scheme_id: @params[:default_light_on_theme] ? @color_scheme.id : nil,
+        )
+      end
+      if @params.has_key?(:default_dark_on_theme)
+        Theme.find_default.update!(
+          dark_color_scheme_id: @params[:default_dark_on_theme] ? @color_scheme.id : nil,
+        )
+      end
```

Comment:
I think we can be more efficient here by only executing one database statement to update regardless of the params.


```suggestion
      update_opts = {}

      if @params.has_key?(:default_light_on_theme)
        update_opts[:color_scheme_id] = @params[:default_light_on_theme] ? @color_scheme.id : nil
      end

      if @params.has_key?(:default_dark_on_theme)
        update_opts[:dark_color_scheme_id] = @params[:default_dark_on_theme] ? @color_scheme.id : nil
      end

      Theme.find_default.update!(update_opts)
```

Reply:
Reading through this again, I'm wondering if this logic should belong in this service given that its sole responsibility is to revise a color scheme. This block of code is actually revising a theme.

Reply:
Yeah, fair, I moved it to controller level. At this stage, defining new service feels a bit a premature.